### PR TITLE
Lock all versions to "jammy" (no "noble" auto-upgrade)

### PR DESCRIPTION
**Pull request by tianon**
Attempts to upgrade 3.x versions to Ubuntu Noble fail because they still require python2 (which is no longer available in Noble).

Attempts to upgrade 4.x fail because of incompatibilities with Python 3.12 (which is the version Noble has).

Attempts to upgrade 5.x fail because of the above Python 3.12 incompatibility, but with the added benefit of code detecting that we're trying to use Python 3.12 and erroring that no suitable Python version could be found (instead of failing with an obscure `six.moves` import error, like 4.x does).

For now, this is an explicit override for every supported version because hope springs eternal that Python 3.12 will be supported eventually and some future 5.x will be able to upgrade to Ubuntu Noble. 🙈

### CODE REVIEW

**Comment by tianon** [comment]
We've also got this fun little warning on 4.1+ (https://github.com/apache/cassandra/commit/917d74bd35712f8374ecefd1d890d02916162daa) several times while running our tests:

> Warning: Using a password on the command line interface can be insecure.
> Recommendation: use the credentials file to securely provide the password.

https://github.com/docker-library/cassandra/actions/runs/10203255159/job/28229194787?pr=284#step:7:15

It's a little bit annoying to fix because the two options for doing so (`--credentials` or `--insecure-password-without-warning`) were _also_ both added only in 4.1+, so we couldn't reasonably squash that warning in our tests without some light version/capability detection.

(Not a serious issue, just something I wanted to note down _somewhere_ and this PR where I had to test all the versions is as good a place as any.)

File: versions.sh [hunk 1]
```diff
@@ -49,10 +62,11 @@ for version in "${versions[@]}"; do
...
 	export javaVersion="${javaVersions[$version]:-$defaultJavaVersion}"
+	suiteOverride="${suiteOverrides[$version]:-}"
 
 	# for the given Java version, find the "default" Eclipse Temurin tag with stable specificity ("X-jre-SUITE")
 	from="$(
-		bashbrew --arch amd64 list --arch-filter "https://github.com/docker-library/official-images/raw/HEAD/library/eclipse-temurin:$javaVersion-jre" \
+		bashbrew --arch amd64 list --arch-filter "https://github.com/docker-library/official-images/raw/HEAD/library/eclipse-temurin:$javaVersion-jre${suiteOverride:+-$suiteOverride}" \
 			| grep -F ":$javaVersion-jre-" \
 			| tail -1
 	)"
```
Comment (yosifkit) [inline versions.sh:72]: Perhaps we should double check that `from` has a value in order to catch the case where a cassandra version is stuck on an Ubuntu that `eclipse-temurin` no longer supports.

Comment (tianon) [reply versions.sh:72]: Ah good call -- I did it this way (with keeping the round trip through `bashbrew list` even though the answer _should_ be literally `eclipse-temurin:XXX-jre-YYY` that we could've generated without going through this) specifically for being able to make sure the image we're trying to use actually does exist and is still supported :smile:

Comment (tianon) [reply versions.sh:72]: Oh, that case is already handled by `set -e` (and `set -o pipefail`) because `bashbrew list` will explicitly non-zero exit. :+1:

Comment (tianon) [reply versions.sh:72]: To illustrate:

```console
$ git --no-pager diff
diff --git a/versions.sh b/versions.sh
index 6532c13..b40ffc8 100755
--- a/versions.sh
+++ b/versions.sh
@@ -22,7 +22,7 @@ declare -A suiteOverrides=(
 	# https://github.com/apache/cassandra/commit/8fd44ca8fc9e0b0e94932bcd855e2833bf6ca3cb#diff-8d8ae48aaf489a8a0e726d3e4a6230a26dcc76e7c739e8e3968e3f65c995d148
 	# https://issues.apache.org/jira/browse/CASSANDRA-19245?focusedCommentId=17803539#comment-17803539
 	# https://github.com/apache/cassandra/blob/cassandra-5.0-rc1/bin/cqlsh#L65
-	[5.0]='jammy'
+	[5.0]='foobar'
 )
 
 cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
$ ./update.sh 5.0
failed fetching repo "https://github.com/docker-library/official-images/raw/HEAD/library/eclipse-temurin:17-jre-foobar"
tag not found in manifest for "eclipse-temurin": "17-jre-foobar"
$ echo $?
1
```

**Review by yosifkit — APPROVED** [review]

**Pull request merged**